main.go

```diff
@@ -38,13 +38,16 @@ import (
 
 func main() {
 	var cliConfig string
+	var APIGroup string
 
 	flagSet := flag.NewFlagSet("cluster-machine-approver", flag.ExitOnError)
 
 	klog.InitFlags(flagSet)
 	flagSet.StringVar(&cliConfig, "config", "", "CLI config")
 	flagSet.Parse(os.Args[1:])
 
+	flagSet.StringVar(&APIGroup, "apigroup", "machine.openshift.io", "API group for machines, defaults to machine.openshift.io")
```

Comment:
i'm curious what you think about making this an environment variable instead of flag. my reason for asking is that we [currently use the `CAPI_GROUP` env variable in the autoscaler](https://github.com/kubernetes/autoscaler/blob/2afb7ab9e1b8e10933e509ed8888a1a0014ff650/cluster-autoscaler/cloudprovider/clusterapi/clusterapi_controller.go#L302) for similar functionality. if we made this an env variable as well, it would be the same mechanism we use for the autoscaler.

not a blocker for me, but i'm curious if we should try to impose some similar behavior here.